### PR TITLE
Add heater

### PR DESCRIPTION
**Pull request by JELoohuis**
- **Added support for the heater category of devices.**
Since my test device had limited functionality the following capabilities were implemented and tested:
  - onoff (switch)
  - target_temperature (temp_set)
  - measure_temperature (temp_current)
  - measure_power (work_power)

  The eco-mode (mode_eco) and child_lock (lock) capability were added untested,
  so it would be nice to get feedback on them from users with devices that support them.

  The level instruction could not be tested and was not added,
  as it is unclear whether all devices that support it use all three levels,
  which would dictate the type of Homey capability used.

May resolve:
- https://github.com/athombv/com.tuya/issues/10
- https://github.com/athombv/com.tuya/issues/16

### CODE REVIEW

**Comment by MerelHudepohl** [comment]
@JELoohuis i'm gonna put this in test so we might get some feedback. Once the icons are available please add them.

File: drivers/heater/device.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    if (typeof status['lock'] === 'boolean') {
+      this.setCapabilityValue('child_lock', status['lock']).catch(this.error);
+    }
+
+    if (typeof status['work_power'] === 'number') {
+      const cur_power = status['cur_power'] / 10.0;
+      this.setCapabilityValue('measure_power', cur_power).catch(this.error);
+    }
```
Comment (wouter-athom) [inline drivers/heater/device.js:52]: Does the presence of 'work_power' guarantee the presence of 'cur_power'?

File: drivers/heater/device.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      value: !!value,
+    });
+  }
+
+  async ecoModeCapabilityListener(value) {
+    await this.sendCommand({
+      code: 'mode_eco',
+      value: value,
+    })
+  }
```
Comment (wouter-athom) [inline drivers/heater/device.js:86]: You used a boolean cast for the other listeners, but not for this one.

File: lib/TuyaOAuth2Driver.js [hunk 3]
```diff
@@ -12,12 +12,16 @@ class TuyaOAuth2Driver extends OAuth2Driver {
...
       .filter(device => {
         this.log('Device:', JSON.stringify(device));
         return this.onTuyaPairListDeviceFilter(device);
-      })
-      .map(device => ({
+      });
+    const listDevices = [];
+    for (const device of filteredDevices) {
+      const deviceSpecs = await oAuth2Client.getSpecification({deviceId: device.id});
+      const deviceProperties = this.onTuyaPairListDeviceProperties({...device}, deviceSpecs);
```
Comment (wouter-athom) [inline lib/TuyaOAuth2Driver.js:23]: These deviceSpecs are never used. Is this for a future update?

File: drivers/heater/device.js (hunk 1 above)
Comment (JELoohuis) [reply drivers/heater/device.js]: This was a typo, fixed in 6c6b0c27802292f381c65ba86c9b73383dc6e85d

File: drivers/heater/device.js (hunk 2 above)
Comment (JELoohuis) [reply drivers/heater/device.js:86]: The redundant casts were removed in bf758dde21f712c84ebf1343fe1a04347f682dfa

File: lib/TuyaOAuth2Driver.js (hunk 3 above)
Comment (JELoohuis) [reply lib/TuyaOAuth2Driver.js:23]: The specifications are used to set the temperature range supported by the device in eb47c39af132eae3aee1555ae4011f54faa33590

File: drivers/heater/driver.flow.compose.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        {
+          "name": "value",
+          "type": "checkbox",
+          "title": { "en": "Value" }
+        }
+      ]
+    }
+  ],
+  "conditions": [],
+  "triggers": []
```
Comment (bobvandevijver) [inline drivers/heater/driver.flow.compose.json:39]: We probably also want triggers for the child lock and eco mode capabilities, as those can be changed on the device as well.

File: drivers/heater/driver.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    this.homey.flow
+      .getActionCard("heater_set_eco_mode")
+      .registerRunListener(async (args) => {
+        await args.device.triggerCapabilityListener("eco_mode", args.value);
+      });
+  }
+
+  static TUYA_DEVICE_CATEGORIES = [
+    TuyaOAuth2Constants.DEVICE_CATEGORIES.SMALL_HOME_APPLIANCES.HEATER,
+  ];
```
Comment (bobvandevijver) [inline drivers/heater/driver.js:26]: Let's always put this as the first code in the driver class, so it can be spotted easily.

**Review by bobvandevijver — CHANGES_REQUESTED** [review]

File: drivers/heater/driver.flow.compose.json (hunk 4 above)
Comment (JELoohuis) [reply drivers/heater/driver.flow.compose.json]: Added in a95db003b642934471f0578717ba7ae2fb077a8e

**Review by bobvandevijver — APPROVED** [review]

**Pull request merged**